File: 20. Valid Parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    s = s.split('')
    let ans = []
    let obj = {
        '(':')',
        '[':']',
        '{':'}'
    }
    for(let i=0; i<s.length; i++){
        if(')]}'.includes(s[i])){
            if(obj[ans.pop()] != s[i])
                return false
        } else{
            ans.push(s[i])
        }
    }
    return ans.length ? false : true
};

File: 20-ValidParentheses/20-ValidParentheses.py
# Last updated: 6/22/2026, 12:54:56 PM
class Solution:
    def isValid(self, s: str) -> bool:
        ans = []
        obj = {
            '(':')',
            '[':']',
            '{':'}'
        }
        for i in s:
            if(i in ')]}'):
                if(len(ans)):
                    if(obj[ans.pop()] != i):
                        return False
                else: return False
            else: ans.append(i)
        return False if len(ans) else True